runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1450,8 +1454,62 @@ NNFW_STATUS nnfw_session::train_export_circle(const char *path)
     return NNFW_STATUS_INVALID_STATE;
   }
 
-  // NYI
-  return NNFW_STATUS_ERROR;
+  class MMappedFile
+  {
+  public:
+    MMappedFile(const char *filename) { _fd = open(filename, O_RDWR); }
+
+    bool ensure_mmap()
+    {
+      struct stat file_stat;
+      if (fstat(_fd, &file_stat) != 0)
+        return false;
+      // file_stat.st_size is `off_t` while msync, munmap requires `size_t`
+      // `off_t` may be long or long long.
+      // `size_t` may be uint32 or uint64.
+      static_assert(sizeof(off_t) == sizeof(size_t), "sizeof(off_t) != sizeof(size_t)");
```

Comment:
As far as I know, we can know `bool-constexpr` without message of `static_assert.

Reply:
Yes, I made the message more meaningful. Also, I've updated `cond` because it is okay that `size_t` have bigger size than `off_t`.